ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/bootstrap/WorkspaceComponent.java

```diff
@@ -35,84 +38,135 @@
             @Override
-            public void apply(List<UsersWorkspaceDto> arg) throws OperationException {
-                if (!arg.isEmpty()) {
-                    operationInfo.setStatus(Status.FINISHED);
-                    Config.setCurrentWorkspace(arg.get(0));
-                    appContext.setWorkspace(arg.get(0));
-                    callback.onSuccess(WorkspaceComponent.this);
-                } else {
+            public void apply(List<UsersWorkspaceDto> workspaces) throws OperationException {
+                operationInfo.setStatus(Status.FINISHED);
+
+                if (workspaces.isEmpty()) {
                     createWorkspacePresenter.show(operationInfo, callback);
+
+                    return;
                 }
+
+                for (UsersWorkspaceDto workspace : workspaces) {
+                    if (RUNNING.equals(workspace.getStatus())) {
+                        setCurrentWorkspace(operationInfo, workspace);
+
+                        return;
+                    }
+                }
+
+                startWorkspacePresenter.show(workspaces, callback, operationInfo);
             }
         }).catchError(new Operation<PromiseError>() {
             @Override
             public void apply(PromiseError arg) throws OperationException {
                 operationInfo.setStatus(Status.ERROR);
+                loader.show(operationInfo);
+
                 callback.onFailure(new Exception(arg.getCause()));
             }
         });
     }
 
-    public void startWorkspace(String id, String envName, final Callback<Component, Exception> callback) {
-        final OperationInfo startWsOperation =
-                new OperationInfo(locale.startingOperation("workspace"), Status.IN_PROGRESS, loader);
+    private void setCurrentWorkspace(OperationInfo operationInfo, UsersWorkspaceDto workspace) {
+        operationInfo.setStatus(Status.FINISHED);
+        Config.setCurrentWorkspace(workspace);
+        appContext.setWorkspace(workspace);
+        callback.onSuccess(WorkspaceComponent.this);
+    }
+
+    public void startWorkspace(String id, final String envName) {
+        final OperationInfo startWsOperation = new OperationInfo(locale.startingOperation("workspace"), Status.IN_PROGRESS, loader);
+
         loader.print(startWsOperation);
+
         workspaceServiceClient.startById(id, envName).then(new Operation<UsersWorkspaceDto>() {
             @Override
-            public void apply(UsersWorkspaceDto arg) throws OperationException {
-                startWsOperation.setStatus(Status.FINISHED);
-                Config.setCurrentWorkspace(arg);
-                appContext.setWorkspace(arg);
-                callback.onSuccess(WorkspaceComponent.this);
+            public void apply(UsersWorkspaceDto workspace) throws OperationException {
+                messageBus = messageBusProvider.createMessageBus(workspace.getId());
```

Comment:
MessageBusProvider#createMessageBus(String) method should receive WebSocket URL but you pass workspace ID.
Isn't it mistake?